lib/public/utilities/DataSource.js

```diff
@@ -0,0 +1,117 @@
+/**
+ *  @license
+ *  Copyright CERN and copyright holders of ALICE O2. This software is
+ *  distributed under the terms of the GNU General Public License v3 (GPL
+ *  Version 3), copied verbatim in the file "COPYING".
+ *
+ *  See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ *  In applying this license CERN does not waive the privileges and immunities
+ *  granted to it by virtue of its status as an Intergovernmental Organization
+ *  or submit itself to any jurisdiction.
+ */
+import { Observable } from '/js/src/index.js';
+import { ObservableData } from './ObservableData.js';
+
+/**
+ * Class to abstract notifying from data changes
+ * @template T - type of data served by data source
+ */
+export class DataSource extends Observable {
+    /**
+     * Constructor
+     */
+    constructor() {
+        super();
+        this._dataSource = null;
+
+        /**
+         * Produce default (initial) value stored in ObservableData
+         * @return {*} default
+         */
+        this._default = () => null;
```

Comment:
There is no notion of default value, even in the observable data. Default is value to use when none other is present, which is different from `initial value`.
In any case, datasource provide a pipe (observable data) which is not a default value.